addon-test-support/-private/filter-test-modules.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Return the matched test.
+ * e.g. if an input is '!/weight/' it returns an array, ['!/weight/', '!', 'weight', ''];
+ *
+ * @param {*} modulePathFilter
+ */
+function getRegexFilter(modulePathFilter) {
+  return /^(!?)\/([\w\W]*)\/(i?$)/.exec( modulePathFilter );
+}
+
+/**
+ * Return a list of test modules that contain a given module path string.
+ *
+ * @param {*} modules
+ * @param {*} modulePathFilter
+ */
+function stringFilter(tests, modules, modulePathFilter) {
+  modules.forEach((module) => {
```

Comment:
We can use filter and return the filtered modules, this way we just return a new array with the filtered modules, instead of modifying the state of the tests array.. 
```
function stringFilter(modules, modulePathFilter) {
  return modules.filter( module => module.includes(modulePathFilter) );
}
```
Then we can simplify filterTestModules as well:
```
function filterTestModules(modules, modulePathFilter) {
  const modulePathRegexFilter = getRegexFilter(modulePathFilter);

  if (modulePathRegexFilter) {
    return regexFilter(tests, modules, modulePathRegexFilter);
  } 
  return stringFilter(tests, modules, modulePathFilter);
} 
```